.github/workflows/deployment.yml

```diff
@@ -0,0 +1,268 @@
+      - staging
+      - production
+
+    paths:
+      - ".github/workflows/deployment.yml"
+      - "portal/**"
+      - "gef-ui/**"
+      - "trade-finance-manager-ui/**"
+      - "portal-api/**"
+      - "dtfs-central-api/**"
+      - "trade-finance-manager-api/**"
+      - "external-api/**"
+      - "azure-functions/acbs-function/**"
+      - "azure-functions/number-generator-function/**"
+
+env:
+  PRODUCT: dtfs
+  ENVIRONMENT: ${{ github.ref_name }}
+  TIMEZONE: "Europe/London"
+  # Base artifact
+  FROM: latest
+
+jobs:
+  setup:
+    name: Setup 🔧
+    runs-on: [self-hosted, linux, deployment]
+    outputs:
+      product: ${{ env.PRODUCT }}
+      environment: ${{ env.ENVIRONMENT }}
+      timezone: ${{ env.TIMEZONE }}
+    steps:
+      - name: Environment 🧪
+        run: echo "Environment set to ${{ env.ENVIRONMENT }}"
+
+      - name: Timezone 🌐
+        run: echo "Timezone set to ${{ env.TIMEZONE }}"
+
+  webapp:
+    name: WebApp 🌐
+    needs: setup
+    environment: ${{ needs.setup.outputs.environment }}
+    runs-on: [self-hosted, DTFS, deployment]
+
+    env:
+      ENVIRONMENT: ${{ needs.setup.outputs.environment }}
+
+    strategy:
+      max-parallel: 2
+      # Do not cancel in-progress jobs upon failure
+      fail-fast: false
+      # Single dimension matrix
+      matrix:
+        webapp:
+          [
+            "portal-ui",
+            "gef-ui",
+            "trade-finance-manager-ui",
+            "portal-api",
+            "dtfs-central-api",
+            "external-api",
+            "trade-finance-manager-api",
+          ]
+
+    concurrency:
+      group: deployment-webapp-${{ github.workflow }}-${{ github.workflow_ref }}-${{ matrix.webapp }}
+      cancel-in-progress: true
+
+    steps:
+      - name: Repository 🗃️
+        uses: actions/checkout@v4
+
+      - name: Defaults ✨
+        uses: Azure/cli@v1.0.9
+        with:
+          inlineScript: |
+            # Basic
+            az configure --defaults location=${{ vars.REGION }}
+            az configure --defaults group=${{ secrets.RESOURCE_GROUP }}
+
+      - name: Azure 🔐
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: CLI 📝
+        run: |
+          echo ACR=$(az acr show -n $(az resource list --resource-type 'Microsoft.ContainerRegistry/registries' --query '[0].name' -o tsv) --query loginServer -o tsv) >> $GITHUB_ENV
+          echo ACR_USER=$(az acr show -n $(az resource list --resource-type 'Microsoft.ContainerRegistry/registries' --query '[0].name' -o tsv) --query name -o tsv) >> $GITHUB_ENV
+          echo WEBAPP=$(az resource list --resource-type 'Microsoft.Web/sites' --query '[?contains(name, `${{ matrix.webapp }}`)].name' -o tsv) >> $GITHUB_ENV
+
+      - name: ACR 🔐
+        uses: azure/docker-login@v1
+        with:
+          login-server: ${{ env.ACR }}
+          username: ${{ env.ACR_USER }}
+          password: ${{ secrets.ACR_PASSWORD }}
+
+      - name: Directory 📁
```

Comment:
At some point should we just update the name of the directory and microservice to portal-ui? :D (not for this PR ofc)